build/test-aot-compatibility.ps1

```diff
@@ -10,6 +10,14 @@ foreach ($line in $($publishOutput -split "`r`n"))
     if ($line -like "*analysis warning IL*") 
     {
         Write-Host $line
+
+        # OpenTelemetry.Instrumentation.SqlClient is properly AOT-annotated.
+        # The caller of OpenTelemetry.Instrumentation.SqlClient will be notified.
+        if ($line -like "*Trimming is not yet supported with SqlClient*")
```

Comment:
Should we just remove this from the TestApp?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/64df5e3df8db55e4da482e577c3524b53213f4b2/test/OpenTelemetry.AotCompatibility.TestApp/OpenTelemetry.AotCompatibility.TestApp.csproj#L25

Reply:
Actually, these warnings are coming from callers of the SqlClient APIs, not from inside the SqlClient library. Since the library's APIs are annotated with `RequiresUnreferencedCode` you shouldn't be getting warnings.

We should remove this code from the script and annotate any callers as `RequiresUnreferencedCode` as well.

Reply:
I've properly annotated the callers; thank you.